src/librustc_resolve/lib.rs

```diff
@@ -1030,7 +1016,7 @@ pub struct Resolver<'a, 'tcx: 'a> {
 
     graph_root: Module<'a>,
 
-    trait_item_map: FnvHashMap<(Name, DefId), DefId>,
+    trait_item_map: FnvHashMap<(Name, DefId), bool /* is static method? */>,
```

Comment:
I'd prefer to have a Static/Instance (or something) enum here, but I don't feel too strongly.